### ISSUE TITLE
eslint showing error when using variable from parent scope

### ISSUE THREAD

**Issue by afucher** [eslint/eslint#12004]
**Tell us about your environment**

* **ESLint Version:** 6.0.1
* **Node Version:** v10.15.3
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default 
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import React from 'react';

interface Props {
    readonly history: number[];
}

export const MyHistory = (props: Props) => {
    const renderContent = () => {
        if (props.history.length > 0) {
            return <div> My history here</div>
        }
        return <div> No historical events</div>
    };

    return renderContent();
};
```

```bash
npm run eslint --fix --ext .tsx --ext .ts src
```

**What did you expect to happen?**
eslint show no errors

**What actually happened? Please include the actual, raw output from ESLint.**
```
  9:19  error  'history' is missing in props validation         react/prop-types
  9:27  error  'history.length' is missing in props validation  react/prop-types
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by platinumazure** [comment]
Hi @afucher, thanks for the issue.

This seems to be an issue with [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react). Would you mind [opening a new issue](https://github.com/yannickcr/eslint-plugin-react/issues/new) there?
